### PR TITLE
Editor socket connection

### PR DESCRIPTION
**Pull request by lucko**
Now in a workable state so opening up for others to have a look over the changes and test.

Still not committed to the idea, especially if the consensus is that we're uneasy with regards to security.

To test, you'll need to use one of the plugin jars from the https://github.com/LuckPerms/LuckPerms/tree/feature/editor-socket branch.

### CODE REVIEW

File: src/util/ws.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    name: 'RSASSA-PKCS1-v1_5',
+    modulusLength: 4096,
+    publicExponent: new Uint8Array([1, 0, 1]),
+    hash: 'SHA-256',
+  }, true, ['sign']);
+
+  const encodedPublicKey = await exportKey('spki', publicKey, 'editor-public-key');
+  const encodedPrivateKey = await exportKey('pkcs8', privateKey, 'editor-private-key');
+
+  console.log('Generated fresh keys');
```
Comment (Aberdeener) [inline src/util/ws.js:52]: Intentionally left in?

Comment (lucko) [reply src/util/ws.js]: oops, thanks!

**Comment by lucko** [comment]
UX needs some work still

- Would be good to display an indication that the editor is connected - maybe a little green tick in the top corner?
- some sort of indication that the save was successful. The modal is a bit too intrusive, don't want something that the user needs to close (the idea is you can seamlessly keep editing), but again, maybe a little green tick, or a smaller pop-up modal that fades after a while
- if the plugin responds saying that the editor is untrusted, we should open a modal to prompt the user to go back in-game and trust it (see the TODO in the socket js file)

Would appreciate help with this from anyone who enjoys vue/frontend dev more than me 😅

**Comment by lucko** [comment]
Ready for a more thorough review.

**Pull request merged**